{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import AlignIO, SeqIO, Align\n",
    "import pandas as pd\n",
    "import os, re, sys, shutil, subprocess\n",
    "from copy import deepcopy\n",
    "\n",
    "# IMPORTANT: EDIT THE PARAMETERS BELOW BEFORE RUNNING!\n",
    "# ------------------------------------------------------------------------------------------------\n",
    "# The directory where this notebook will create files and work from.\n",
    "# The directory should either:\n",
    "#   (a) be empty\n",
    "#   (b) contain a ./input directory you pre-populated with fastas containing the genes to align (one per organism)\n",
    "cwd = '/home/ecutts/Bacteroidetes/rprot-concatenations/rprot-aln-all-chlorobi/'\n",
    "\n",
    "overwrite_input = True\n",
    "# If True, preexisting files in ./input will be DELETED and replaced with files from input_directories\n",
    "\n",
    "input_ext = '.fasta'\n",
    "# Extension of inputs. Files in input_directories ending in input_ext will be copied to ./input\n",
    "\n",
    "input_directories = ['/home/ecutts/Bacteroidetes/Refseq-blast/rprots/all/', \\\n",
    "                     '/home/ecutts/Bacteroidetes/MAGs-blast/rprots/edited', \\\n",
    "                    '/home/ecutts/Bacteroidetes/Chlorobi/rprots/']\n",
    "# Directories (absolute path) containing inputs (can be left blank if ./input was pre-populated)\n",
    "\n",
    "map_names = True\n",
    "# Map sequence names to filenames, useful for mapping taxa to filenames\n",
    "# If True, edit the function get_name_from_file() to work for your file naming scheme\n",
    "# see below for example\n",
    "\n",
    "genes = {} # see below, I wrote code to generate this automatically\n",
    "# Dictionary mapping gene names (for files) to search terms (for text searching)\n",
    "# Search terms should be specific (i.e. will only return one gene in your fastas)\n",
    "\n",
    "\n",
    "# EXTRA CODE SPECIFIC TO YOUR FILES AND GENES\n",
    "# ------------------------------------------------------------------------------------------------\n",
    "# Populate the gene_search_terms dict with conserved rprots\n",
    "for g in pd.read_table('/home/ecutts/metadata/rprots.txt', header=None)[0].tolist():\n",
    "    genes[g] = ' ' + g + ' [' # \"L1\" can also show up in \"L13\", so this makes the search terms specific.\n",
    "    \n",
    "# get_name_from_taxa()\n",
    "# change \n",
    "from ete3 import NCBITaxa\n",
    "ncbi = NCBITaxa()\n",
    "\n",
    "def get_name_from_file(file):\n",
    "    if 'rprots_MAG' in file: # If file is a Shark Bay MAG\n",
    "        name = file.split('rprots_')[1].split('.fasta')[0]\n",
    "        return name.replace('_', ' ') # Return MAG ##\n",
    "    else: # else, the file is a NCBI refseq genome with a genus, species name\n",
    "        taxid = int(file.split('_')[1])\n",
    "        genus_species = ncbi.translate_to_names([taxid])[0]\n",
    "        return(str(taxid) + ' ' + genus_species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(cwd):\n",
    "    os.makedirs(cwd)\n",
    "os.chdir(cwd)\n",
    "if os.path.exists('./input') and overwrite_input == True:\n",
    "    shutil.rmtree('./input')\n",
    "os.makedirs('./input', exist_ok=True)\n",
    "os.makedirs('./pre-aln', exist_ok=True) # ./gene-pre-aln will contain the pre-aligned files for each gene\n",
    "os.makedirs('./aln', exist_ok=True) # ./gene-aln will contain the individual alignments for each gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of input files: 323\n"
     ]
    }
   ],
   "source": [
    "# Import files from input_directories ending in input_ext\n",
    "preexisting = [f for f in os.listdir('./input') if f.endswith(input_ext)]\n",
    "for d in input_directories:\n",
    "    for f in os.listdir(d):\n",
    "        if f not in preexisting and f.endswith(input_ext):\n",
    "            shutil.copy2(os.path.join(d,f), './input')\n",
    "            \n",
    "inputs = [f for f in os.listdir('./input') if f.endswith(input_ext)] # only use files with extension input_ext\n",
    "print('Number of input files: ' + str(len(inputs)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "for g in genes.keys():\n",
    "    search_term = genes[g]\n",
    "    with open('pre-aln/' + g + '.fasta', 'w') as outfile:\n",
    "        for f in inputs:\n",
    "            file = './input/' + f\n",
    "            if map_names == True:\n",
    "                name = get_name_from_file(file)\n",
    "            else:\n",
    "                name = f\n",
    "            with open(file, 'r') as infile:\n",
    "                for record in SeqIO.parse(infile, 'fasta'):\n",
    "                    if search_term in record.description:\n",
    "                        record.id = name\n",
    "                        record.description = \"\"\n",
    "                        record.name = name\n",
    "                        SeqIO.write(record, outfile, 'fasta')       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checksum: # of sequences in each pre-align file\n",
      "If all genomes have all genes, checksum = 323\n",
      "\n",
      "L1.fasta: 320 ***\n",
      "L2.fasta: 320 ***\n",
      "L3.fasta: 321 ***\n",
      "L4.fasta: 322 ***\n",
      "L5.fasta: 318 ***\n",
      "L6.fasta: 319 ***\n",
      "L10.fasta: 321 ***\n",
      "L13.fasta: 322 ***\n",
      "L14.fasta: 318 ***\n",
      "L15.fasta: 319 ***\n",
      "L18.fasta: 321 ***\n",
      "L22.fasta: 318 ***\n",
      "L23.fasta: 322 ***\n",
      "L24.fasta: 318 ***\n",
      "L29.fasta: 315 ***\n",
      "S2.fasta: 321 ***\n",
      "S3.fasta: 320 ***\n",
      "S4.fasta: 321 ***\n",
      "S5.fasta: 321 ***\n",
      "S7.fasta: 318 ***\n",
      "S8.fasta: 319 ***\n",
      "S9.fasta: 322 ***\n",
      "S10.fasta: 321 ***\n",
      "S11.fasta: 320 ***\n",
      "S12.fasta: 319 ***\n",
      "S13.fasta: 319 ***\n",
      "S14.fasta: 319 ***\n",
      "S15.fasta: 320 ***\n",
      "S17.fasta: 317 ***\n",
      "S19.fasta: 321 ***\n"
     ]
    }
   ],
   "source": [
    "print('Checksum: # of sequences in each pre-align file')\n",
    "print('If all genomes have all genes, checksum = ' + str(len(os.listdir('input'))))\n",
    "print()\n",
    "check = len(os.listdir('input'))\n",
    "for f in os.listdir('./pre-aln'):\n",
    "    checksum = 0\n",
    "    with open('pre-aln/' + f, 'r') as checkfile:\n",
    "        for record in SeqIO.parse(checkfile, 'fasta'):\n",
    "            checksum +=1\n",
    "    if checksum != check:   \n",
    "        print(f + ': ' + str(checksum) + ' ***')\n",
    "    else:\n",
    "        print(f + ': ' + str(checksum))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing and duplicate genes:\n",
      "\n",
      "search term: ' L1 [' - 0 in rprots_496057_GCF_007971385.1.fasta\n",
      "search term: ' L2 [' - 0 in rprots_496057_GCF_007971385.1.fasta\n",
      "search term: ' L3 [' - 0 in rprots_496057_GCF_007971385.1.fasta\n",
      "search term: ' L5 [' - 0 in rprots_496057_GCF_007971385.1.fasta\n",
      "search term: ' S11 [' - 0 in rprots_496057_GCF_007971385.1.fasta\n",
      "search term: ' S7 [' - 0 in rprots_1189621_GCF_000265075.1.fasta\n",
      "search term: ' S15 [' - 0 in rprots_1189621_GCF_000265075.1.fasta\n",
      "search term: ' L5 [' - 0 in rprots_2518971_GCF_004803915.1.fasta\n",
      "search term: ' L6 [' - 0 in rprots_2518971_GCF_004803915.1.fasta\n",
      "search term: ' L15 [' - 0 in rprots_1605367_GCF_004329705.1.fasta\n",
      "search term: ' S2 [' - 0 in rprots_2049305_GCF_006974145.1.fasta\n",
      "search term: ' L22 [' - 0 in rprots_1912246_GCF_008085455.1.fasta\n",
      "search term: ' L2 [' - 0 in rprots_MAG_15.fasta\n",
      "search term: ' L14 [' - 0 in rprots_MAG_15.fasta\n",
      "search term: ' L22 [' - 0 in rprots_MAG_15.fasta\n",
      "search term: ' L24 [' - 0 in rprots_MAG_15.fasta\n",
      "search term: ' L29 [' - 0 in rprots_MAG_15.fasta\n",
      "search term: ' S3 [' - 0 in rprots_MAG_15.fasta\n",
      "search term: ' S17 [' - 0 in rprots_MAG_15.fasta\n",
      "search term: ' S19 [' - 0 in rprots_MAG_15.fasta\n",
      "search term: ' L29 [' - 0 in rprots_MAG_42.fasta\n",
      "search term: ' S17 [' - 0 in rprots_MAG_42.fasta\n",
      "search term: ' S4 [' - 0 in rprots_MAG_45.fasta\n",
      "search term: ' S11 [' - 0 in rprots_MAG_45.fasta\n",
      "search term: ' S13 [' - 0 in rprots_MAG_45.fasta\n",
      "search term: ' S7 [' - 0 in rprots_MAG_55.fasta\n",
      "search term: ' S12 [' - 0 in rprots_MAG_55.fasta\n",
      "search term: ' L13 [' - 0 in rprots_MAG_56.fasta\n",
      "search term: ' L15 [' - 0 in rprots_MAG_56.fasta\n",
      "search term: ' S2 [' - 0 in rprots_MAG_56.fasta\n",
      "search term: ' S9 [' - 0 in rprots_MAG_56.fasta\n",
      "search term: ' L5 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' L6 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' L14 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' L15 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' L18 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' L24 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' L29 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' S5 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' S7 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' S8 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' S12 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' S13 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' S14 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' S17 [' - 0 in rprots_MAG_69.fasta\n",
      "search term: ' L1 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L3 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L4 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L5 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L10 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L14 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L22 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L23 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L24 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L29 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' S7 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' S8 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' S10 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' S12 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' S14 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' S17 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' S19 [' - 0 in rprots_MAG_81.fasta\n",
      "search term: ' L6 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' L14 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' L24 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' L29 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' S3 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' S7 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' S8 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' S10 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' S12 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' S14 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' S15 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' S17 [' - 0 in rprots_MAG_83.fasta\n",
      "search term: ' L1 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L5 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L6 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L10 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L14 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L15 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L18 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L22 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L24 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L29 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S3 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S4 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S5 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S8 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S11 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S13 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S14 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S15 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' S17 [' - 0 in rprots_MAG_84.fasta\n",
      "search term: ' L2 [' - 0 in rprots_319225_GCF_000012485.1.fasta\n",
      "search term: ' L29 [' - 0 in rprots_290315_GCF_000020465.1.fasta\n",
      "search term: ' L22 [' - 0 in rprots_517417_GCF_000020505.1.fasta\n",
      "search term: ' L29 [' - 0 in rprots_324925_GCF_000020645.1.fasta\n",
      "search term: ' S13 [' - 0 in rprots_1974213_GCF_002113825.1.fasta\n"
     ]
    }
   ],
   "source": [
    "# This block will identify the files that have \"problems\" with the checksum above\n",
    "# Comment out or skip if your checksum is good.\n",
    "print('Missing and duplicate genes:')\n",
    "print()\n",
    "for file in os.listdir('./input'):\n",
    "    genedict = dict.fromkeys(genes.values(), 0)\n",
    "    with open('input/' + file, 'r') as readfile:\n",
    "        for record in SeqIO.parse(readfile, 'fasta'):\n",
    "            for gene in genedict.keys():\n",
    "                if gene in record.description:\n",
    "                    genedict[gene] += 1\n",
    "    for gene in genedict.keys():\n",
    "        if genedict[gene] != 1:\n",
    "            print('search term: ' + \"'\" + gene + \"' - \"  + str(genedict[gene]) + ' in ' + file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "mafft = '/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft' # path to mafft\n",
    "\n",
    "def align(file): # function that performs alignment and prints process info\n",
    "    output = open('./aln/' + file + '.aln', 'w')\n",
    "    print(\n",
    "        subprocess.run([mafft, '--reorder', '--auto', './pre-aln/' + file], stdout=output)\n",
    "    )\n",
    "    output.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L1.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L2.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L3.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L4.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L5.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L6.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L10.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L13.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L14.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L15.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L18.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L22.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L23.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L24.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/L29.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S2.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S3.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S4.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S5.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S7.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S8.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S9.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S10.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S11.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S12.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S13.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S14.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S15.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S17.fasta'], returncode=0)\n",
      "CompletedProcess(args=['/cm/shared/engaging/mafft/7.245-with-extensions/bin/mafft', '--reorder', '--auto', './pre-aln/S19.fasta'], returncode=0)\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir('./pre-aln'): #create alignments of individual gene files\n",
    "    align(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# INDIVIDUAL ALIGNMENT COMPLETE. NOW, CONCATENATE ALIGNMENTS.\n",
    "aln_extension = '.aln'\n",
    "genomes = {}\n",
    "for aln in os.listdir('./aln'):\n",
    "    if not aln.endswith(aln_extension):\n",
    "        continue\n",
    "    alignment = AlignIO.read('./aln/' + aln, 'fasta')\n",
    "    genomes[aln] = set()\n",
    "    for entry in alignment:\n",
    "        genome = entry.description\n",
    "        if genome in genomes[aln]:\n",
    "            print('\\t**Error, duplicated genome in %s: %s' %(aln, genome))\n",
    "            break\n",
    "        genomes[aln].add(genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_union = set.union(*genomes.values())\n",
    "missing_genes = {} #keeps track of # of missing marker genes in each genome\n",
    "concatenation = {}\n",
    "for genome in genome_union:\n",
    "    missing_genes[genome] = 0\n",
    "    concatenation[genome] = Align.SeqRecord(Align.Seq(''))\n",
    "    concatenation[genome].name = genome\n",
    "    concatenation[genome].id = genome\n",
    "    concatenation[genome].description = genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "L1.fasta\n",
      "L2.fasta\n",
      "L3.fasta\n",
      "L4.fasta\n",
      "L5.fasta\n",
      "L6.fasta\n",
      "L10.fasta\n",
      "L13.fasta\n",
      "L14.fasta\n",
      "L15.fasta\n",
      "L18.fasta\n",
      "L22.fasta\n",
      "L23.fasta\n",
      "L24.fasta\n",
      "L29.fasta\n",
      "S2.fasta\n",
      "S3.fasta\n",
      "S4.fasta\n",
      "S5.fasta\n",
      "S7.fasta\n",
      "S8.fasta\n",
      "S9.fasta\n",
      "S10.fasta\n",
      "S11.fasta\n",
      "S12.fasta\n",
      "S13.fasta\n",
      "S14.fasta\n",
      "S15.fasta\n",
      "S17.fasta\n",
      "S19.fasta\n"
     ]
    }
   ],
   "source": [
    "total_genes = 0.0\n",
    "current_position = 1\n",
    "partitions = open('partitions.txt', 'w')\n",
    "for aln in os.listdir('./aln'):\n",
    "    if not aln.endswith(aln_extension):\n",
    "        continue\n",
    "    print(aln.replace(aln_extension, ''))\n",
    "    tmp_aln = AlignIO.read('./aln/' + aln, 'fasta')\n",
    "    aln_length = tmp_aln.get_alignment_length() #expected alignment size\n",
    "    total_genes += aln_length\n",
    "    \n",
    "    for entry in tmp_aln:\n",
    "        if len(entry) != aln_length:\n",
    "            print('\\t**Error, block \"%s\" has a different length than the rest of the MSA: %s'\\\n",
    "                 %(entry.name, aln))\n",
    "        genome = entry.description\n",
    "        concatenation[genome] += deepcopy(entry.seq)\n",
    "        \n",
    "    partitions.write('LG, %s = %i-%i\\n' %(aln.replace(aln_extension, ''),\\\n",
    "                                          current_position, current_position+aln_length-1))\n",
    "    current_position += aln_length\n",
    "    \n",
    "    for genome in genome_union.difference(genomes[aln]):\n",
    "        concatenation[genome] += Align.Seq( '-' * aln_length)\n",
    "        missing_genes[genome] += aln_length\n",
    "partitions.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AlignIO.write(Align.MultipleSeqAlignment(concatenation.values()), 'concatenated_alignment.aln', 'fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
